test/common/ssl/ssl_socket_test.cc

```diff
@@ -60,9 +62,10 @@ void testUtil(const std::string& client_ctx_json, const std::string& server_ctx_
 
   Json::ObjectSharedPtr client_ctx_loader = TestEnvironment::jsonLoadFromString(client_ctx_json);
   ClientContextConfigImpl client_ctx_config(*client_ctx_loader);
-  ClientContextPtr client_ctx(manager.createSslClientContext(stats_store, client_ctx_config));
-  Network::ClientConnectionPtr client_connection = dispatcher.createSslClientConnection(
-      *client_ctx, socket.localAddress(), Network::Address::InstanceConstSharedPtr());
+  Ssl::ClientSslSocketFactory ssl_socket_factory(client_ctx_config, manager, stats_store);
+  Network::ClientConnectionPtr client_connection = dispatcher.createClientConnection(
```

Comment:
entirely optional but I like to take advantage of changes like this to move move members to the test class so future PRs require less cut and paste changes.